modular_zubbers/code/game/objects/items/fuel_flamer.dm

```diff
@@ -0,0 +1,125 @@
+
+/obj/item/gun/fuel_thrower
+	name = "fuel flamethrower"
+	icon_state = "flamethrowerbase"
+	inhand_icon_state = "flamethrower_0"
+	resistance_flags = FIRE_PROOF
+	icon = 'icons/obj/weapons/flamethrower.dmi'
+	lefthand_file = 'icons/mob/inhands/weapons/flamethrower_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/flamethrower_righthand.dmi'
+	dry_fire_sound = null
+	fire_sound = null
+	suppressed_sound = null
+	force = 3
+	throwforce = 10
+	throw_speed = 1
+	throw_range = 3
+	light_on = FALSE
+	light_system = OVERLAY_LIGHT
+	light_color = LIGHT_COLOR_FLARE
+	light_range = 2
+	light_power = 2
+	custom_materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT)
+	var/lit = FALSE
+	var/max_fuel = 100
+	var/fuel_consumption = 5
+	var/datum/reagent/fuel_type = /datum/reagent/fuel
+
+/obj/item/gun/fuel_thrower/Initialize(mapload)
+	. = ..()
+	desc = "A flamethrower that uses [lowertext(initial(fuel_type.name))] as ammunition."
+	chambered = new /obj/item/ammo_casing/flamer(src)
+	create_reagents(max_fuel)
+	reagents.add_reagent(fuel_type, max_fuel)
+	update_appearance(ALL)
+
+/obj/item/gun/fuel_thrower/update_icon_state()
+	inhand_icon_state = "flamethrower_[lit]"
+	return ..()
+
+/obj/item/gun/fuel_thrower/update_overlays()
+	. = ..()
+	. += "+igniter1"
+	. += "+ptank"
```

Comment:
Why aren't these suffixes in there by default if update_overlays is going to add them unconditionally anyways?

Reply:
Because I believed that update_overlays cuts all overlays, thus would remove them if update_overlays was called

Reply:
Wait what type is `.` I must have misread lol, I thought this was just modifying icon_state

Reply:
. isn't a type, . is the default return variable